src/initialization/InitDistribution.cpp

```diff
@@ -121,6 +125,91 @@ namespace impactx
         }
     }
 
+    void initialization::set_distribution_parameters_from_twiss_inputs (
+        amrex::ParmParse const & pp_dist,
+        amrex::ParticleReal& sigx, amrex::ParticleReal& sigy, amrex::ParticleReal& sigt,
+        amrex::ParticleReal& sigpx, amrex::ParticleReal& sigpy, amrex::ParticleReal& sigpt,
+        amrex::ParticleReal& muxpx, amrex::ParticleReal& muypy, amrex::ParticleReal& mutpt
+    )
+    {
+
+        // Values to be read from input
+        amrex::ParticleReal betax, betay, betat, emittx, emitty, emittt;
+        // If alpha is zero the bunch is in focus
+        amrex::ParticleReal alphax = 0.0, alphay = 0.0, alphat = 0.0;
+
+        // Reading the input Twiss parameters
+        pp_dist.query("alphaX", alphax);
+        pp_dist.query("alphaY", alphay);
+        pp_dist.query("alphaT", alphat);
+        pp_dist.get("betaX", betax);
+        pp_dist.get("betaY", betay);
+        pp_dist.get("betaT", betat);
+        pp_dist.get("emittX", emittx);
+        pp_dist.get("emittY", emitty);
+        pp_dist.get("emittT", emittt);
+
+        if (betax <= 0.0_prt || betay <= 0.0_prt || betat <= 0.0_prt) {
+            throw std::runtime_error("Input Error: The beta function values need to be non-zero positive values in all dimensions.");
+        }
+
+        if (emittx <= 0.0_prt || emitty <= 0.0_prt || emittt <= 0.0_prt) {
+            throw std::runtime_error("Input Error: Emittance values need to be non-zero positive values in all dimensions.");
+        }
+
+        std::array<amrex::ParticleReal, 3> const alphas = {alphax, alphay, alphat};
+        std::array<amrex::ParticleReal, 3> const betas = {betax, betay, betat};
+        std::array<amrex::ParticleReal, 3> const emittances = {emittx, emitty, emittt};
+
+        // calculate Twiss / Courant-Snyder gammas
+        amrex::Vector<amrex::ParticleReal> gammas;
+        for (size_t i = 0; i < alphas.size(); i++)
+            gammas.push_back((1.0 + pow(alphas.at(i), 2)) / betas.at(i));
```

Comment:
You want this: https://en.cppreference.com/w/cpp/numeric/math/pow
```suggestion
            gammas.push_back((1.0 + std::pow(alphas.at(i), 2)) / betas.at(i));
```

Otherwise it is the C function for `double`:
https://en.cppreference.com/w/c/numeric/math/pow